src/plugins/analysis/file_type/code/file_type.py

```diff
@@ -24,11 +28,17 @@ class Schema(pydantic.BaseModel):
         )
 
     def __init__(self):
+        try:
+            version_file = Path(get_bin_dir()) / 'version.json'
+            fw_magic_db_version = json.loads(version_file.read_text()).get('version')
+        except (json.JSONDecodeError, FileNotFoundError):
+            fw_magic_db_version = None
         super().__init__(
             metadata=AnalysisPluginV0.MetaData(
                 name='file_type',
                 description='identify the file type',
-                version='1.0.0',
+                version=Version(1, 0, 1),
+                system_version=fw_magic_db_version,
```

Comment:
Isn't this technically wrong?
While the plugin defensively depends on the version of the  8firmware-magic-database ](https://github.com/fkie-cad/firmware-magic-database), it also depends on the version of file.